ui/src/app/services/sign-in.service.ts

```diff
@@ -52,6 +52,7 @@ export class SignInService {
           this.subscribeToAuth2User();
         });
         resolve(gapi.auth2);
+        this.router.initialNavigation();
```

Comment:
Perhaps this belongs somewhere more central like app component, if you can hook into that promise there. Feels a bit off that navigation is kicked off in `sign-in.service`, and also within a method that could seemingly be called more than once.